packages/embark-deploy-tracker/src/deploymentChecks.js

```diff
@@ -0,0 +1,85 @@
+import {__} from 'embark-i18n';
+import {toChecksumAddress} from 'embark-utils';
+import Web3 from "web3";
+
+export default class DeploymentChecks {
+  constructor({trackingFunctions, logger, events, plugins}) {
+    this.trackingFunctions = trackingFunctions;
+    this.logger = logger;
+    this.events = events;
+    this.plugins = plugins;
+    this._web3 = null;
+  }
+
+  get web3() {
+    return (async () => {
+      if (!this._web3) {
+        const provider = await this.events.request2("blockchain:client:provider", "ethereum");
+        this._web3 = new Web3(provider);
+      }
+      return this._web3;
+    })();
+  }
+
+  async checkContractConfig(params, cb) {
+    const {contract} = params;
+
+    // previous event action check
+    if (!params.shouldDeploy) {
+      return cb(null, params);
+    }
+
+    // contract config address field set - do not deploy
+    if (contract.address !== undefined) {
+      try {
+        toChecksumAddress(contract.address);
+      } catch (e) {
+        return cb(e);
+      }
+      contract.deployedAddress = contract.address;
+      contract.log(contract.className.bold.cyan + __(" already deployed at ").green + contract.deployedAddress.bold.cyan);
```

Comment:
The goal was to keep things DRY as we log contract info in a few different places, and each time need to check on `contract.silent` before logging. Was thinking to extend the method to accept a log level as a parameter, but this PR was big enough, and it can be handled in the future if needed.